### PR TITLE
update to python 3

### PR DESCRIPTION
**Pull request by decarlof**
This should address #1

### CODE REVIEW

**Comment by decarlof** [comment]
I might need help before considering merging the above. 

There is an error in the summary table generation that I might have introduced by the Python 2->3 update. When using:

.. idl:autopath:: ../path/
	:summary: 

I get this:

Screenshot 2024-02-23 at 17 48 36
	
In other words the summary table is built correctly for *.pro containing function, but fails for *.pro containing pro. 

[handle_idl_object_table](https://github.com/alexrudy/sphinx-idl/blob/978122d10a01caa1a834bcb5ac5c789f4dc83cc5/sphinx_idl/auto.py#L99) seems to create correct signature_line, summary_line. 
- For function:
`:func:`ucomp_julday2dateobs` Convert a Julian date into DATE-OBS string of the form "YYYY-MM-DDTHH:MM:SSZ"`
- For pro:
`:pro:`ucomp_ut2hst` Convert an HST datetime to UT.`

but the [IDLAutoPath.run()](https://github.com/alexrudy/sphinx-idl/blob/978122d10a01caa1a834bcb5ac5c789f4dc83cc5/sphinx_idl/auto.py#L178) returns the following nodes and body:

- For function:
`py-func">ucomp_julday2dateobs()</literal></pending_xref></paragraph></entry><entry><paragraph>Convert a Julian date into DATE-OBS string of the form "YYYY-MM-DDTHH:MM:SSZ"`
- For pro:
`</paragraph></entry></row><row><entry><paragraph><problematic ids="id2" refid="id1">:pro:`ucomp_ut2hst`</problematic></paragraph></entry><entry><paragraph>Convert an HST datetime to UT.`

I assume this is not what happens under Python 2. 
Do you have any insight on what may cause this?

thanks

**Comment by alexrudy** [comment]
Thanks so much for this contribution, this is awesome! Let me take a look and see if I can figure out what is going on - its been a minute since I dug into the implementation here.

**Comment by alexrudy** [comment]
I think I've figured out your issue! By default, sphinx uses the `py` domain, so `:func:` is really `:py:func:` (and is detected, but doesn't link anywhere) and `:pro:` is unknown, but `:idl:pro:` is probably recognized.

We can fix the `autopath` table generation in a different PR so that it emits `:idl:pro:` by default and not `:pro:` (same for `:func:`).

I usually set `primary_domain = "idl"` in my `conf.py` when working on IDL projects, so that I don't have to prefix directives and roles with `idl:`. (though doing so isn't a problem)

File: sphinx_idl/domain.py
Comment (alexrudy) [inline sphinx_idl/domain.py:1]: On line 289, the call to `.items()` changes behavior between python2 and python3 - in python2 it returns a list of items, in python3 it returns a view. The view will break if the underlying dictionary is modified. Therefore, we should wrap the `self.data['objects'].items()` in `list(self.data['objects'].items())`.

**Review by alexrudy — APPROVED** [review]
Overall, this looks good to me, and I'm happy to merge as is.

It has raised two follow up issues that I'll create:

1. The auto-stuff should emit fully-qualified directives and roles (i.e. with the `idl:` prefix).
2. The note I left about the bug in `clear_doc()` below. This isn't a show stopper, you just have to run `make clean` before `make html` each time.

**Pull request merged**